whirlpool/appliancesmanager.py

```diff
@@ -108,27 +135,75 @@ async def _get_shared_appliances(self) -> bool:
 
             return True
 
-    async def fetch_appliances(self):
+    async def fetch_appliances(self) -> bool:
         account_id = await self._auth.get_account_id()
         if not account_id:
             return False
+
+        # Invalidate cached property
+        self.__dict__.pop("all_appliances", None)
+
         success_owned = await self._get_owned_appliances(account_id)
         success_shared = await self._get_shared_appliances()
 
         return success_owned or success_shared
 
-    @property
-    def aircons(self):
-        return self._aircons
-
-    @property
-    def refrigerators(self):
-        return self._refrigerators
+    async def fetch_all_data(self):
+        for appliance in self.all_appliances.values():
+            await appliance.fetch_data()
+
+    async def connect(self):
+        """Connect to appliance event listener"""
+        await self.start_event_listener()
+
+    async def disconnect(self):
+        """Disconnect from appliance event listener"""
+        await self.stop_event_listener()
+
+    async def start_event_listener(self):
+        """Start the appliance event listener"""
+        await self.fetch_all_data()
+        if self._event_socket is not None:
+            LOGGER.warning("Event socket not None when starting event listener")
+
+        self._event_socket = EventSocket(
+            await self._getWebsocketUrl(),
+            self._auth,
+            self.all_appliances.keys(),
+            self._event_socket_callback,
+            self.fetch_all_data,
+            self._session,
+        )
+        self._event_socket.start()
+
+    async def stop_event_listener(self):
+        """Stop the appliance event listener"""
+        await self._event_socket.stop()
+        self._event_socket = None
+
+    def _event_socket_callback(self, msg: str):
+        json_msg = json.loads(msg)
+        said = json_msg["said"]
+        app = self.all_appliances().get(said)
+        if app is None:
+            LOGGER.warning("Received message for unknown appliance %s", said)
+            return
+        app._update_appliance_attributes(
```

Comment:
`_update_appliance_attributes` does not exist. should it be `update_attributes`?